proposals/minty-in-the-browser.md

```diff
@@ -0,0 +1,159 @@
+# Minty in the browser - Mint NFTs end to end inside a decentralized web app using js-ipfs
+
+Authors: @yusefnapora
+
+Initial PR: [#92](https://github.com/protocol/web3-dev-team/pull/92) 
+
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+
+In mid-March 2021, based on [this earlier proposal](https://github.com/protocol/web3-dev-team/pull/11) we published a guide on the IPFS docs side called [Mint an NFT with IPFS](https://docs.ipfs.io/how-to/mint-nfts-with-ipfs/), which walks through an example app called [minty](https://github.com/yusefnapora/minty) and dissects the code to show how everything works.
+
+In the interests of simplicity and shipping quickly, minty is a command line Node.js app and can't run in its current form inside a web browser. This proposal outlines the work required to make minty work end-to-end in an ethereum-enabled browser (e.g. with the MetaMask extension).
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+
+- D-app developers want to mint NFTs using IPFS without relying on a centralized backend server
+  - This may not actually be the case. I'm not aware of any NFT platform today that's "fully decentralized" in the sense of doing everything on chain without a backend API.
+
+- We want to encourage d-app devs to use js-ipfs directly in their d-apps, as opposed to using an HTTP api from Pinata, nft.storage, etc
+  - This seems like a good thing to encourage, but it also raises the bar for adoption a bit, since js-ipfs is a heavier dependency than an HTTP client and may require learning new concepts, etc.
+  
+- D-app devs are willing to use a remote pinning service for persistence
```

Comment:
how different is the experience for a dapp creator if they use a remote backend server (a la infura) or a remote pinning service (a la pinata)? I guess the difference is using an HTTP API vs using IPFS directly to parse/upload the file client side?

Reply:
I don't think it's a huge difference, since you'll probably want to use a pinning service either way.

If you were running a backend server, you could potentially pin your data to your own IPFS nodes instead of using a pinning service, as long as those nodes are well-connected and up all the time. But since browser nodes can't really be up all the time, it seems like the pinning service is closer to "required" than "recommended".

There's probably a way to do it without a pinning service that I'm not thinking of though... hopefully someone will think of it :)